Docs around null handling for unwrapped single-field schemas. (#2944)
confluentinc/ksql: docs/developer-guide/serialization.rst

default, defined by ``ksql.persistence.wrap.single.values``. You can change the system default.
For more information, see :ref:`ksql-persistence-wrap-single-values`.

> .. important:: KSQL treats ``null` keys and values as a special case. It is recommended that

Reviewer: ```suggestion
.. important:: KSQL treats ``null` keys and values as a special case. We recommend avoiding
```

>                unwrapped single-field schemas are avoided if the field can have a ``null`` value.

Reviewer: ```suggestion
               unwrapped single-field schemas if the field can have a ``null`` value.
```

A ``null`` value in a table's topic is treated as a tombstone, used to indicate a row has been
removed. If a table's source topic has an unwrapped single-field key schema and the value is

> ``null``, it be treated as a tombstone, resulting in any previous value for the key being removed

Reviewer: ```suggestion
``null``, it's treated as a tombstone, resulting in any previous value for the key being removed
```

from the table.

> A ``null`` key or value in a stream's topic will be ignored when the stream is part of a join.

Reviewer: ```suggestion
A ``null`` key or value in a stream's topic is ignored when the stream is part of a join.
```

> A ``null`` key in a table's topic is treated as a tombstone, and a ``null`` value is

Reviewer 2: Did you mean the null key is ignored and the null value is tombstone?

Author: Good catch. Thanks.

> A ``null`` value in a table's topic is treated as a tombstone, and a ``null`` key is

Reviewer 3: move this to the previous paragraph

> A ``null`` key in a table's topic will be treated as a tombstone and a ``null` value will be

Reviewer: ```suggestion
A ``null`` key in a table's topic is treated as a tombstone, and a ``null`` value is
```

> ignored, when the table is part of a join,

Reviewer: ```suggestion
ignored when the table is part of a join.
```

> Care should be taken when dealing with unwrapped single-field schemas to ensure any ``null`` key or

Reviewer: ```suggestion
When you have an unwrapped single-field schema, ensure that any ``null`` key or
```

default, defined by ``ksql.persistence.wrap.single.values``. You can change the system default.
For more information, see :ref:`ksql-persistence-wrap-single-values`.

> .. important:: KSQL treats ``null` keys and values as a special case. It is recommended that

Reviewer: ```suggestion
.. important:: KSQL treats ``null` keys and values as a special case. We recommended avoiding
```

>                unwrapped single-field schemas are avoided if the field can have a ``null`` value.

Reviewer: ```suggestion
               unwrapped single-field schemas if the field can have a ``null`` value.
```

> A ``null`` value in a table's topic is treated as a tombstone, used to indicate a row has been

Reviewer: ```suggestion
A ``null`` value in a table's topic is treated as a tombstone, which indicates that a row has been
```

removed. If a table's source topic has an unwrapped single-field key schema and the value is

> ``null``, it be treated as a tombstone, resulting in any previous value for the key being removed

Reviewer: ```suggestion
``null``, it's treated as a tombstone, resulting in any previous value for the key being removed
```

from the table.

> A ``null`` key or value in a stream's topic will be ignored when the stream is part of a join.

Reviewer: ```suggestion
A ``null`` key or value in a stream's topic is ignored when the stream is part of a join.
```

> A ``null`` key in a table's topic is treated as a tombstone, and a ``null`` value is

Reviewer 2: Sam here, did you mean the null key is ignored and the null value is tombstone?

> A ``null`` key in a table's topic will be treated as a tombstone and a ``null` value will be

Reviewer: ```suggestion
A ``null`` key in a table's topic is treated as a tombstone, and a ``null`` value is
```

> ignored, when the table is part of a join,

Reviewer: ```suggestion
ignored when the table is part of a join,
```

> Care should be taken when dealing with unwrapped single-field schemas to ensure any ``null`` key or

Reviewer: When you have an unwrapped single-field schema, ensure that any ``null`` key or
